Source/Core/Core/IOS/Network/KD/Mail/WC24FriendList.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright 2023 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "Core/IOS/Network/KD/Mail/WC24FriendList.h"
+#include "Core/IOS/FS/FileSystem.h"
+#include "Core/IOS/Uids.h"
+
+namespace IOS::HLE::NWC24::Mail
+{
+WC24FriendList::WC24FriendList(std::shared_ptr<FS::FileSystem> fs) : m_fs{std::move(fs)}
+{
+  ReadFriendList();
+}
+
+void WC24FriendList::ReadFriendList()
+{
+  const auto file = m_fs->OpenFile(PID_KD, PID_KD, FRIEND_LIST_PATH, FS::Mode::Read);
+  if (!file || !file->Read(&m_data, 1))
+    return;
+
+  const s32 file_error = CheckFriendList();
```

Comment:
Should this be a `bool`?  Also maybe instead say `const bool success` since you check for the opposite when posting the error.

Reply:
Yes it should be.